kodilla-patterns2/src/main/java/com/kodilla/patterns2/adapter/bookclasifier/MedianAdapter.java

```diff
@@ -0,0 +1,21 @@
+package com.kodilla.patterns2.adapter.bookclasifier;
+
+import com.kodilla.patterns2.adapter.bookclasifier.librarya.Book;
+import com.kodilla.patterns2.adapter.bookclasifier.librarya.Classifier;
+import com.kodilla.patterns2.adapter.bookclasifier.libraryb.BookSignature;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class MedianAdapter extends MedianAdaptee implements Classifier {
+
+    @Override
+    public int publicationsYearMedian(Set<Book> bookSet) {
+        Map<BookSignature, com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book> books = new HashMap<>();
+        for (Book book : bookSet) {
```

Comment:
Widzę, że preferujesz `for` od streamów ;)
Jeśli chcesz, możesz w formie dodatkowego ćwiczenia przepisać to na stream.